### PR TITLE
Fix / ETH gas not accounted in profit

### PR DESCRIPTION
**Pull request by Nullably**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**After creating this PR, please make sure:**

- [ ] You link the related GitHub issue or ticket

**A description of the changes proposed in the pull request**:
- Add quote-ETH price fetch loop in amm_arb 
- The price is then used to convert ETH gas amount into quote value for profitability calculation

### CODE REVIEW

**Review by vic-en — APPROVED** [review]
LGTM

**Review by sdgoh — DISMISSED** [review]
LGTM

File: hummingbot/strategy/amm_arb/data_types.py [hunk 1]
```diff
@@ -58,9 +59,24 @@ def profit_pct(self, account_for_fee: bool = False) -> Decimal:
...
+                                                                 sell.amount)
+
+        # buy_fee_amount = buy_trade_fee.fee_amount_in_quote(buy.market_info.trading_pair,
+        #                                                    buy.quote_price, buy.amount)
+        # sell_fee_amount = sell_trade_fee.fee_amount_in_quote(sell.market_info.trading_pair, sell.quote_price,
+        #                                                      sell.amount)
+
         sell_gained_net = (sell.amount * sell.quote_price) - sell_fee_amount
         buy_spent_net = (buy.amount * buy.quote_price) + buy_fee_amount
         return (sell_gained_net - buy_spent_net) / buy_spent_net if buy_spent_net != s_decimal_0 else s_decimal_nan
```
Comment (vic-en) [inline hummingbot/strategy/amm_arb/data_types.py:82]:
```suggestion
        return (sell_gained_net - buy_spent_net) / buy_spent_net if buy_spent_net != s_decimal_0 else s_decimal_0
```

File: hummingbot/strategy/amm_arb/data_types.py [hunk 2]
```diff
@@ -58,12 +59,27 @@ def profit_pct(self, account_for_fee: bool = False) -> Decimal:
...
+
+        # buy_fee_amount = buy_trade_fee.fee_amount_in_quote(buy.market_info.trading_pair,
+        #                                                    buy.quote_price, buy.amount)
+        # sell_fee_amount = sell_trade_fee.fee_amount_in_quote(sell.market_info.trading_pair, sell.quote_price,
+        #                                                      sell.amount)
+
         sell_gained_net = (sell.amount * sell.quote_price) - sell_fee_amount
         buy_spent_net = (buy.amount * buy.quote_price) + buy_fee_amount
-        return (sell_gained_net - buy_spent_net) / buy_spent_net if buy_spent_net != s_decimal_0 else s_decimal_nan
+        return (sell_gained_net - buy_spent_net) / buy_spent_net if buy_spent_net != s_decimal_0 else s_decimal_0
```
Comment (vic-en) [inline hummingbot/strategy/amm_arb/data_types.py:82]:
```suggestion
        return ((sell_gained_net - buy_spent_net) / buy_spent_net) if buy_spent_net != s_decimal_0 else s_decimal_0
```

**Review by mifeng — APPROVED** [review]
While there are some issues, they appear limited to pairs that include USDC and WBTC

**Review by sdgoh — APPROVED** [review]
LGTM

**Pull request merged**